### ISSUE TITLE
Cannot Update picture for products!

### ISSUE THREAD

**Issue by InzamamAnwar** [Abdur-rahmaanJ/shopcube#510]
I cloned the repository and started server as described. However, I cannot update the product picture, as it is showing the following error

```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: resources.category_id
[SQL: INSERT INTO resources (filename, type, category, created_date, product_id, category_id, subcategory_id) VALUES (?, ?, ?, ?, ?, ?, ?)]
[parameters: ('7da36728_6_image_1_cropped.jpg', 'image', 'product_image', '2022-12-02 02:14:21.828496', 1, None, None)]

```

Similarly while creating new Sub-category it is giving following error

```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: resources.category_id

```

I think in the **Resource**, model having **category_id** nullable=**True** is main culprit. But before assigning it **False** I want to confirm whether it will break other parts or not?

**Comment by Abdur-rahmaanJ** [comment]
@InzamamAnwar Let me investigate. Thanks for reporting!

**Comment by Abdur-rahmaanJ** [comment]
@InzamamAnwar Fixed in e56da8ed0e173b0efa99201b44a5c606ea5a57ee in 4.3.1

**Comment by Abdur-rahmaanJ** [comment]
Btw we need to work on the tests ^^/ to answer your question fully!
